context.go

```diff
@@ -0,0 +1,69 @@
+package api2go
+
+import (
+	"golang.org/x/net/context"
+	"time"
+)
+
+// APIContext api2go context for handlers, nil implementations related to Deadline and Done.
+type APIContext struct {
```

Comment:
If I haven't missed anything, apiContext could be private and use an public Interface right? Since the user only implements a middleware with it being the first parameter.